Added warning message for #1105

taichi-ishitani opened the pull request:
Hi,

I added warning message for #1105 as a special failure message.

JonRowe commented on lib/rspec/matchers/built_in/be_instance_of.rb:
```diff
@@ -5,16 +5,71 @@ module BuiltIn
+          if actual_object_respond_to?(actual, :instance_of?)
+            actual.instance_of?(expected)
+          else
+            @no_checker_implemented = true
```
Rather than storing the result of the check, I suggest that you raise an `ArgumentError` here with the warning message and shortcut the rest.

JonRowe commented on lib/rspec/matchers/built_in/be_kind_of.rb:
```diff
@@ -5,10 +5,67 @@ module BuiltIn
+          elsif actual_object_respond_to?(actual, :is_a?)
+            actual.is_a?(expected)
+          else
+            @no_checkers_implemented = true
```
Rather than storing the result of the check, I suggest that you raise an `ArgumentError` here with the warning message and shortcut the rest.

taichi-ishitani commented:
Hi @JonRowe 
Thanks for your feedback and I implemented your feedback.
Also I added examples for following cases
- actual object does not respond to #kind_of? method
- actual object does not respond to #is_a? method

JonRowe commented on lib/rspec/matchers/built_in/be_instance_of.rb:
```diff
@@ -14,7 +14,21 @@ def description
+          ::Kernel.instance_method(:respond_to?).bind(actual).call(method)
+        end
+
+        def error_message
```
Given this is now a raise and reused, can we inline it?

taichi-ishitani replied:
Yes, I will update.

taichi-ishitani replied:
@JonRowe 
You mean that `#actual_object_respond_to?` should be inline?
If not, can you provide me more details?

JonRowe replied:
```
  raise ArgumentError, "The #{matcher_name} matcher requires that " \
                                      "the actual object responds to #instance_of? method " \
                                      "but it does not respond to the method."
```

taichi-ishitani replied:
Thank you!

taichi-ishitani replied:
Done.

JonRowe commented on lib/rspec/matchers/built_in/be_kind_of.rb:
```diff
@@ -8,7 +8,23 @@ class BeAKindOf < BaseMatcher
+          ::Kernel.instance_method(:respond_to?).bind(actual).call(method)
+        end
+
+        def error_message
```
Given this is now a raise and reused, can we inline it?
